### ISSUE TITLE
Blank white screen when running archlinux

### ISSUE THREAD

**Issue by z3roco01** [logisim-evolution/logisim-evolution#1235]
I am trying to run logisim-evolution with `./gradlew run` on archlinux and it loads but gives a blank white screen

**Comment by MarcinOrlowski** [comment]
Please provide more information. What branch you try to build, what version it is. It's pretty unlikely that you can have "blank screen" - mind showing the screenshot?

**Comment by MarcinOrlowski** [comment]
Can you please ensure that this is the screenshot really? Because I am pretty skeptical as i.e. at least window decoration should be there. Mind showing whole desktop screenshot (please hide sensitive stuff if you have any there)

**Comment by MarcinOrlowski** [comment]
what java version you use (`java -version)?

**Comment by Unknown-User2** [comment]
i use 17

**Comment by MarcinOrlowski** [comment]
Or can you try nightly? https://github.com/logisim-evolution/logisim-evolution#nightly-builds

**Comment by Unknown-User2** [comment]
both give the same result

**Comment by Unknown-User2** [comment]
i dont use an ide i just use vim and i dont really know how to use strace

**Comment by Unknown-User2** [comment]
@davidhutchens i tried down to 3.5.0 and it still didnt work but it worked before

@MarcinOrlowski what is the invocation line

**Comment by Unknown-User2** [comment]
i used both `java -jar` and through my file manager thunar

**Comment by roman1267** [comment]
@Unknown-User2 I can confirm this is _probably_ not a Logisim issue; rather, it most likely stems from how the JVM interacts with tiling window managers. I am using bspwm on Manjaro, and I had your exact issue. 

After reading through the Arch wiki page that @eippoc linked, I found that [changing the name of my window manager](https://wiki.archlinux.org/title/java#Impersonate_another_window_manager) using the Suckless utility "wmname" solved the problem for me.

**Comment by Ddystopia** [comment]
I have the same issue, 
5.19.10-arch1-1
dwm-6.3

openjdk 18.0.2 2022-07-19
OpenJDK Runtime Environment (build 18.0.2+0)
OpenJDK 64-Bit Server VM (build 18.0.2+0, mixed mode)

**Comment by maehne** [comment]
@Ddystopia: Please check the comments by [eippoc](https://github.com/logisim-evolution/logisim-evolution/issues/1235#issuecomment-947118624) and [roman1267](https://github.com/logisim-evolution/logisim-evolution/issues/1235#issuecomment-1228998331) above. Please provide feedback If these hints help you, too. A PR which reliably fixes this issue in Logisim-evolution is welcome.

**Comment by Ddystopia** [comment]
@maehne Yes, it helped. I've changed name of my window manager to compiz and it works.
`wmname compiz`

**Comment by maehne** [comment]
@eippoc, @roman1267 and @Ddystopia: Could you please try whether setting `_JAVA_AWT_WM_NONREPARENTING=1` environment variable instead of changing the `wmname` also fixes the issue? One way is to issue `'export _JAVA_AWT_WM_NONREPARENTING=1'` in your shell and launch logisim-evolution from it. Another way is to add it to your `~/.xinitrc`.

While checking out the above links and further references (e.g., [JDK-8058197](https://bugs.openjdk.org/browse/JDK-8058197) and https://github.com/Smithay/smithay/issues/389), I did not find hints on how to fix the issue in the Logisim-evolution code base. So, we may need to document the workarounds in our [README.md](https://github.com/logisim-evolution/logisim-evolution/blob/master/README.md).

Are there any better ideas? Help with this issue is appreciated!

**Comment by eippoc** [comment]
@maehne
setting `_JAVA_AWT_WM_NONREPARENTING=1` does work for both jdk8 and jdk17.0.3, that is how I solved this issue.
The other envvar that is suggested by arch wiki `AWT_TOOLKIT=MToolkit` doesn't seem to do anything on my setup (Gentoo, dwm), but may work on other distros.

**Comment by Ddystopia** [comment]
> @eippoc, @roman1267 and @Ddystopia: Could you please try whether setting `_JAVA_AWT_WM_NONREPARENTING=1` environment variable instead of changing the `wmname` also fixes the issue? One way is to issue `'export _JAVA_AWT_WM_NONREPARENTING=1'` in your shell and launch logisim-evolution from it. Another way is to add it to your `~/.xinitrc`.
> 
> While checking out the above links and further references (e.g., [JDK-8058197](https://bugs.openjdk.org/browse/JDK-8058197) and [Smithay/smithay#389](https://github.com/Smithay/smithay/issues/389)), I did not find hints on how to fix the issue in the Logisim-evolution code base. So, we may need to document the workarounds in our [README.md](https://github.com/logisim-evolution/logisim-evolution/blob/master/README.md).
> 
> Are there any better ideas? Help with this issue is appreciated!

Works for me
